### PR TITLE
feature/micro-reddit

### PR DESCRIPTION
**Pull request by Sick-Carlito**
# Dear reviewer

The assignment is complete, the following tasks are done.

- `User` model
- `Link` model (this could also be named `Post`)
- `Comment` model

If there is something missing or wrong, let us know as soon as possible so we can implement the changes quickly. **Thanks for your time**.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,92 @@
-# README
+# Micro Reddit
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+This is a simple version of micro-reddit using data structures necessary to support link submissions and commenting.
 
-Things you may want to cover:
+- Models, Views, and Controllers (MVC)
```
Comment (santiago-rodrig) [inline README.md]: Just models...

File: README.md [hunk 2]
```diff
@@ -1,24 +1,92 @@
...
+- Data Structures & Relationships
```
Comment (santiago-rodrig) [inline README.md]: A guess of the database schema used on the [reddit](https://www.reddit.com/) site.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,92 @@
...
+- Routing
```
Comment (santiago-rodrig) [inline README.md]: No, because we are not using controllers in this project.

File: README.md [hunk 4]
```diff
@@ -1,24 +1,92 @@
...
+- Migrations
+- Views with forms, partials, and helpers
```
Comment (santiago-rodrig) [inline README.md]: Delete this, not related.

File: README.md [hunk 5]
```diff
@@ -1,24 +1,92 @@
...
+- RESTful design
```
Comment (santiago-rodrig) [inline README.md]: Again, not related.

File: README.md [hunk 6]
```diff
@@ -1,24 +1,92 @@
...
+- Adding gems for extra features
```
Comment (santiago-rodrig) [inline README.md]: We did not use extra gems.

File: README.md [hunk 7]
```diff
@@ -1,24 +1,92 @@
...
 
-* Ruby version
+## Getting Started
 
-* System dependencies
+These instructions will help you get started with the project up and running on your local machine for learning purposes.
```
Comment (santiago-rodrig) [inline README.md]: "for learning purposes", delete that.

File: README.md [hunk 8]
```diff
@@ -1,24 +1,92 @@
...
+  gem 'selenium-webdriver'
+  # Easy installation and use of web drivers to run system tests with browsers
+  gem 'webdrivers'
+end
+
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
+```
+
```
Comment (santiago-rodrig) [inline README.md]: Not necessary, the Gemfile is included.

File: README.md [hunk 9]
```diff
@@ -1,86 +1,38 @@
...
-## Build with
+These instructions will help you get started with the project up and running on your local machine.
+
+
+### What you need:
 
 - Ruby 2.6.5
 - Rails 6.0.1
+- Yarn 1.19.2
+- bundler 2.0.2
```
Comment (santiago-rodrig) [inline README.md]: Don't capitalize on the programs. Also, add `sqlite3 3.27.2`.

File: README.md [hunk 10]
```diff
@@ -1,86 +1,38 @@
...
+
+
+## Testing the models
+
+Since the model instances have been created and the database populated, you can perform the following steps to test out the relationship between the models in your console.
+
+- First log in to the console by doing > rails c.
+
+- Using User model as an example, do > User.all to check if there's some data for the model. This shoul return an array of the model's intances.
```
Comment (santiago-rodrig) [inline README.md]: Here you have a typo.
should instead of shoul

File: README.md [hunk 11]
```diff
@@ -1,24 +1,45 @@
...
-* ...
+- First log in to the console by doing > rails c.
+
+- Using User model as an example, do > User.all to check if there's some data for the model. This should return an array of the model's intances.
+
+- To get the first association between users and comments, do > User.first.comments
+
+- You can also do > User.first.links to see the User - links association
+
+- You can also do > Link.last.comments to see the Link - Comment associations
```
Comment (samgaco) [inline README.md]: :100:

Comment (santiago-rodrig) [reply README.md]: Thanks for the feedback!

**Pull request merged**